src/routes/event/list.tsx

```diff
@@ -60,15 +123,28 @@ export default function EventList() {
         </Stack>
         <Stack>
           <List>
-            {allEvents.map((event) => (
-              <ListItem key={event.eventId} disablePadding>
-                <ListItemButton
-                  onClick={() => navigate("/event/" + event.eventId)}
-                >
-                  <ListItemText primary={event.eventName} />
-                </ListItemButton>
-              </ListItem>
-            ))}
+            {allEvents
+              .filter((event) => {
+                if (
+                  event.eventName.includes(searchName) &&
+                  (!selectedLocation ||
+                    selectedLocationEvents.includes(event.eventId)) &&
+                  (!selectedCategory ||
+                    selectedCategoryEvents.includes(event.eventId))
+                ) {
+                  return true;
+                }
+                return false;
+              })
+              .map((event) => (
+                <ListItem key={event.eventId} disablePadding>
+                  <ListItemButton
+                    onClick={() => navigate("/event/" + event.eventId)}
+                  >
+                    <ListItemText primary={event.eventName} />
+                  </ListItemButton>
+                </ListItem>
+              ))}
```

Comment:
This will re-filter everytime the component re-renders, its a good idea to make these filteres in state.
mapping is fine in this case, because the map, has a "key", thats how react knows if it needs to re-render something or not
But for the filter ? it will do it everytime.